### PR TITLE
Add support for reporting if the plan reports changes

### PR DESCRIPTION
**Pull request by EvyBongers**

### CODE REVIEW

File: action.yaml [hunk 1]
```diff
@@ -2,6 +2,10 @@
 name: Terraform
 description: Terraform all-in-one workflow
 inputs:
+  detailed_exit_code:
+    required: false
+    description: Whether to report a detailed exit code
+    default: 'false'
```
Comment (tun0) [inline action.yaml:8]: This name doesn't have any meaning at this level. It reflects the parameter used by the new feature. But it doesn't indicate what the impact would be.

Perhaps better to add something like a `check for changes` in addition to `plan` and `apply`?

Comment (EvyBongers) [reply action.yaml:8]: I'm okay with that. Should we even have it be part of `apply` then? If changes are going to be automatically applied, having a detailed exit code probably doesn't add any value at all. The job could then skip `apply` and `destroy` before reporting final status.

Comment (tun0) [reply action.yaml:8]: It kinda depends on which use-case(s) you're trying to solve here. I can think of several, each possibly requiring different solutions :wink:

Comment (tun0) [reply action.yaml:8]: And looking at the PR this one's mentioned in, perhaps the actual reporting to Slack could be made part of this actions as well? :shrug:

File: index.js [hunk 2]
```diff
@@ -162,39 +175,51 @@ async function terraform(args) {
...
+    const slack = new WebClient(slackBotToken);
+    const tfd = await terraform(['plan', `-lock=${terraformLock}`, `-parallelism=${terraformParallelism}`, '-out=terraform.plan', '-detailed-exitcode'].concat(terraformTargets));
+    switch (tfd.status) {
+    case 0:
+      break;
+    case 2:
+      core.warning('Terraform reported a diff');
+      const result = await slack.chat.postMessage({
+        icon_url: 'https://cdn.icon-icons.com/icons2/2107/PNG/512/file_type_terraform_icon_130125.png',
+        text: `Unapplied changes were detected for ${terraformWorkspace}:\n\`\`\`\n${tfd.stdout}\n\`\`\``,
```
Comment (tun0) [inline index.js:188]: Just the workspace isn't enough, you want repo and working directory listed as well.

**Pull request merged**